capa/features/extractors/dnfile/extractor.py

```diff
@@ -28,43 +28,41 @@ def __init__(self, path: str):
         self.pe: dnfile.dnPE = dnfile.dnPE(path)
 
         # pre-compute these because we'll yield them at *every* scope.
-        self.global_features: List[Tuple[Feature, int]] = []
+        self.global_features: List[Tuple[Feature, Address]] = []
         self.global_features.extend(capa.features.extractors.dotnetfile.extract_file_os(pe=self.pe))
         self.global_features.extend(capa.features.extractors.dotnetfile.extract_file_arch(pe=self.pe))
 
     def get_base_address(self):
-        return 0x0
+        return AbsoluteVirtualAddress(0x0)
 
     def extract_global_features(self):
         yield from self.global_features
 
     def extract_file_features(self):
         yield from capa.features.extractors.dnfile.file.extract_features(self.pe)
 
-    def get_functions(self):
-        # data structure shared across functions yielded here.
-        # useful for caching analysis relevant across a single workspace.
-        ctx = {}
-        ctx["pe"] = self.pe
-
-        for f in get_dotnet_managed_method_bodies(self.pe):
-            setattr(f, "ctx", ctx)
-            yield f
+    def get_functions(self) -> Iterator[FunctionHandle]:
+        for token, f in get_dotnet_managed_method_bodies(self.pe):
+            yield FunctionHandle(address=DNTokenAddress(Token(token)), inner=f, ctx={"pe": self.pe})
 
     def extract_function_features(self, f):
         # TODO
         yield from []
 
-    def get_basic_blocks(self, f):
+    def get_basic_blocks(self, f) -> Iterator[BBHandle]:
         # each dotnet method is considered 1 basic block
-        yield f
+        yield BBHandle(
+            address=f.address,
+            inner=f.inner,
+        )
 
-    def extract_basic_block_features(self, f, bb):
+    def extract_basic_block_features(self, fh, bbh):
         # we don't support basic block features
         yield from []
 
-    def get_instructions(self, f, bb):
-        yield from f.instructions
+    def get_instructions(self, fh, bbh):
+        for insn in bbh.inner.instructions:
+            yield InsnHandle(address=DNTokenOffsetAddress(bbh.address.token, insn.offset - fh.inner.offset), inner=insn)
```

Comment:
```suggestion
            yield InsnHandle(
                address=DNTokenOffsetAddress(bbh.address.token, insn.offset - (fh.inner.offset + fh.inner.header_size)),
                inner=insn,
            )
```
consider methodef header size when calculating instruction offset

Reply:
- [ ] correctly compute instruction offset